test/test_robin_dict.jl

```diff
@@ -0,0 +1,413 @@
+        @test i == Set([1])
+    end
+    @test Set(string(k) for k in keys(d)) == Set(["1","3"])
+end
+
+@testset "find" begin
+    @test findall(isequal(1), RobinDict(:a=>1, :b=>2)) == [:a]
+    @test sort(findall(isequal(1), RobinDict(:a=>1, :b=>1))) == [:a, :b]
+    @test isempty(findall(isequal(1), RobinDict()))
+    @test isempty(findall(isequal(1), RobinDict(:a=>2, :b=>3)))
+
+    @test findfirst(isequal(1), RobinDict(:a=>1, :b=>2)) == :a
+    @test findfirst(isequal(1), RobinDict(:a=>1, :b=>1, :c=>3)) in (:a, :b)
+    @test findfirst(isequal(1), RobinDict()) === nothing
+    @test findfirst(isequal(1), RobinDict(:a=>2, :b=>3)) === nothing
+end
+
+@testset "haskey" begin
+    h = RobinDict(1=>2, 2=>3)
+    @test haskey(h, 1) == true
+    @test haskey(h, 2) == true
+    @test haskey(h, 3) == false
+    @test_throws MethodError haskey(h, "1")
+end
+
+@testset "getkey" begin
+    h = RobinDict(1=>2, 3 => 6, 5=>10)
+    @test getkey(h, 1, 7) == 1
+    @test getkey(h, 4, 6) == 6
+    @test_throws MethodError getkey(h, "1", 8)
+end
+
+@testset "empty" begin
+    h = RobinDict()
+    for i=1:1000
+        h[i] = i+1
+    end
+    length0 = length(h.hashes)
+    empty!(h)
+    @test h.count == 0
+    @test h.maxprobe == 0
+    @test h.idxfloor == 0
+    @test h.totalcost == 0
+    @test length(h.hashes) == length(h.keys) == length(h.vals) == length0
+end
+
+@testset "ArgumentError" begin
+    @test_throws ArgumentError RobinDict(0)
+    @test_throws ArgumentError RobinDict([1])
+    @test_throws ArgumentError RobinDict([(1,2),0])
+end
+
+@testset "empty tuple" begin
+    h = RobinDict(())
+    @test length(h) == 0
+end
+
+@testset "get_idxfloor" begin
+    h = RobinDict()
+    @test get_idxfloor(h) == 0
+    
+    h["a"] = 1
+    h[2] = "b"
+    @test h.idxfloor == get_idxfloor(h)
+    pop!(h)
+    @test h.idxfloor == get_idxfloor(h)
+    pop!(h)
+    @test h.idxfloor == get_idxfloor(h) == 0 
+end
+
+@testset "invariants" begin
+    h1 = RobinDict{Int, Int}()
+    for i in 1:300
+        h1[i] = i
+    end
+    
+    for i in 1:length(h1.keys)
+        if isslotfilled(h1, i)
+            @test hash_key(h1.keys[i]) == h1.hashes[i]
+        end
+    end
+
+    h2 = RobinDict{Float64, Float64}()
+    for i in 1:300
+        h2[rand()] = rand()
+    end
+    
+    for i in 1:length(h2.keys)
+        if isslotfilled(h2, i)
+            @test hash_key(h2.keys[i]) == h2.hashes[i]
+        end
+    end
+
+    h3 = RobinDict{String, Int}()
+    for i in 1:300
+        h3[randstring()] = i
+    end
+    
+    for i in 1:length(h3.keys)
+        if isslotfilled(h3, i)
+            @test hash_key(h3.keys[i]) == h3.hashes[i]
+        end
+    end
+
+    h = RobinDict()
+    for i = 1:500
+        h[i] = i+1
+    end
+    sz = length(h.keys)
+    for i in 1:sz
+        if isslotfilled(h, i)
+            hash = h.hashes[i]
+            des_ind = desired_index(hash, sz)
+            dist = calculate_distance(h, i)
+            @test dist == (i - des_ind)
```

Comment:
Shouldn't this test fail if the desired index is near the end and the actual index wraps around?